lib/common/patchset.c

```diff
@@ -326,35 +326,54 @@ pcmk__xml_create_patchset(const xmlNode *source, xmlNode *target,
     return xml_create_patchset_v2(source, target);
 }
 
+/*!
+ * \internal
+ * \brief Calculate a digest from changed XML and add it to the patchset
+ *
+ * \param[in]     source    XML before applying patchset
+ * \param[in,out] target    XML with changes applied
+ * \param[in,out] patchset  XML patchset
+ *
+ * \note This function assumes a v2 patchset.
```

Comment:
I don't think it does ... It's called for v1, just unconditionally whereas v2 is conditional on with_digest

Reply:
`patchset_process_digest()` did not assume v2, but `pcmk__patchset_process_digest()` does. There are only two callers:
* `crm_diff.c:generate_patch()`
* `cib_utils.c:cib_perform_op()`

Both of them always generate a v2 patchset in supported versions of Pacemaker.

Reply:
This is either a leftover or an oversight. The only two internal callers always generate a v2 patchset. The now-deprecated `patchset_process_digest()` (public function) could send a v1 patch.